pallets/manta-pay/src/test/payment.rs

```diff
@@ -313,155 +314,186 @@ fn initialize_test(id: AssetId, value: AssetValue) {
 /// Tests multiple to_private from some total supply.
 #[test]
 fn to_private_should_work() {
-    let mut rng = OsRng;
-    new_test_ext().execute_with(|| {
-        let asset_id = rng.gen();
-        let total_free_supply = AssetValue(rng.gen());
-        initialize_test(asset_id, total_free_supply + DEFAULT_ASSET_ED);
-        mint_tokens(
-            asset_id,
-            &value_distribution(5, total_free_supply, &mut rng),
-            &mut rng,
-        );
-    });
+    for _ in 0..RANDOMIZED_TESTS_ITERATIONS {
+        let mut rng = OsRng;
+        new_test_ext().execute_with(|| {
+            let asset_id = rng.gen();
```

Comment:
Can be the random number sampled from a range like `(0, u128::MAX)`?

Reply:
I mean the `amount` not the asset id.

Reply:
no, asset-id is still u32.

Reply:
Some of balances can be from (0, u128::MAX) or (0, u64::MAX), depending on the type of integer.